crates/wdk-build/src/resource_compile.rs

```diff
@@ -0,0 +1,217 @@
+    }
+
+    let ver_filetype = "VFT_DRV";
+    let ver_filesubtype = "VFT2_DRV_SYSTEM";
+    let ver_originalfilename = "VER_INTERNALNAME_STR";
+
+    // Create the RC file content
+    let rc_content = format!(
+        r#"#include <windows.h>
+#include <ntverp.h>
+#define	VER_FILETYPE	            {file_type}
+#define	VER_FILESUBTYPE	            {file_subtype}
+#define VER_INTERNALNAME_STR        "{name}"
+#define VER_ORIGINALFILENAME_STR    {original_filename}
+
+#undef VER_FILEDESCRIPTION_STR     
+#define VER_FILEDESCRIPTION_STR "{description}"
+
+#undef  VER_PRODUCTNAME_STR
+#define VER_PRODUCTNAME_STR    VER_FILEDESCRIPTION_STR
+
+#define VER_FILEVERSION        {fileversion},0
+#define VER_FILEVERSION_STR    "{productversion}.0"
+
+#undef  VER_PRODUCTVERSION
+#define VER_PRODUCTVERSION          VER_FILEVERSION
+
+#undef  VER_PRODUCTVERSION_STR
+#define VER_PRODUCTVERSION_STR      VER_FILEVERSION_STR
+
+#define VER_LEGALCOPYRIGHT_STR      {copyright}
+#ifdef  VER_COMPANYNAME_STR
+
+#undef  VER_COMPANYNAME_STR
+#define VER_COMPANYNAME_STR         {companyname}
+#endif
+
+#undef  VER_PRODUCTNAME_STR
+#define VER_PRODUCTNAME_STR    {productname}
+
+#include "common.ver""#,
+        file_type = ver_filetype,
+        file_subtype = ver_filesubtype,
+        original_filename = ver_originalfilename
+    );
+   
+    std::fs::write("resources.rc", rc_content).expect("Unable to write RC file");
+    invoke_rc(&includeargs, rc_exe_rootpath);
+}
+
+// function to invoke RC.exe
+fn invoke_rc(includeargs: &Vec<String>, rc_exe_rootpath: String) {
+
+    let resource_script = "resources.rc";
+    let rc_exe_path = format!("{}\\rc.exe", rc_exe_rootpath);
+    let rc_exe_path = Path::new(&rc_exe_path);
+    if !rc_exe_path.exists() {
+        eprintln!(
+            "Error: rc.exe path does not exist : {}", 
+            rc_exe_path.display()
+        );
+        std::process::exit(1); // Exit with a non-zero status code
+    }
+
+    let mut command = Command::new(rc_exe_path);
+    command.args(includeargs).arg(resource_script);
+    println!("Command executed: {:?}", command);
+    
+    let status = command.status();
+
+    match status {
+        Ok(exit_status) => {
+            if exit_status.success() {
+                println!("Resource compilation successful!");
+                println!("cargo:rustc-link-arg=resources.res");
+            } else {
+                println!("Resource compilation failed.");
+                std::process::exit(1); // Exit with a non-zero status code
+            }
+        }
+        Err(err) => {
+            eprintln!("Error running rc.exe: {}", err);
+            std::process::exit(1); // Exit with a non-zero status code
+        }
+    }
+}
+// function to get package metadata details
+fn get_packagemetadatadetails() -> (String, String, String) {
+    // Run the 'cargo metadata' command and capture its output
+    let path = env::var("CARGO_MANIFEST_DIR").unwrap();
+    let meta = MetadataCommand::new()
```

Comment:
Rather than do additional parsing here, do you think it would be better to bake this into the WDK config struct @ayodejiige ?